source/MaterialXRuntime/RtNodeGraph.h

```diff
@@ -57,6 +57,12 @@ class RtNodeGraph : public RtNode
     /// Return an iterator over the nodes in the graph.
     RtPrimIterator getNodes() const;
 
+    /// Return any associated version string.
+    const RtToken& getVersion() const;
```

Comment:
I'm not sure nodegraphs can have a version. As far as I can see in the spec version is only for nodedefs, so for a nodegraph implementation its version would be the associated nodedef's version.

Reply:
The match for finding an implementation matches by version so if it's a nodegraph it will check version on the nodegraph. Currently nodedef creation only adds version on the nodedef and this will fail.

Reply:
Yep, for a node instance you need to point out which version you want if there are many different once. Since the complete interface is not always given so it's under-specified. Nodegraphs on the other hand point to a specific nodedef by name, and there can only be one version for a named nodedef.